### ISSUE TITLE
Adding novel 'AdaFamily' optimizer 

### ISSUE THREAD

**Issue by hfassold** [pytorch/pytorch#74904]
### 🚀 The feature, motivation and pitch

It could be nice to add the novel 'AdaFamily' optimizer (see [1]) to the already available optimizers under 'torch.optim'.
Disclaimer: I am the author of this paper, so I might be a bit biased  :-)

I know the evaluation in the paper is limited, but it seems to give benefits for both normal-size models (ResNet etc.) and also for compact models (MobileNet etc.). I did also an evaluation with NLP finetuning task which is not in the paper (using the code at [2]), where it brings an accuracy improvement of ~ 0.5 percent compared to using 'AdamW'.

Furthermore, it is actually not a single method, but a 'family' of algorithms, parametrized by parameter 'myu'.
It's implementation is quite simple (there are just a few lines of code changes compared to code for Adam/AdaBelief/AdaMomentum). If something is unclear, just email me (email: hannes.fassold(at)joanneum.at).

-----------------------

[1] "AdaFamily: A family of Adam-like adaptive gradient methods", Fassold, ISPR, 2022
[2] https://github.com/hfassold/nlp_finetuning_adafamily

**Comment by albanD** [comment]
Hi,

Thanks for opening this issue.
We do have some pretty strict rules related to adding new Optimizer/Modules: https://github.com/pytorch/pytorch/wiki/Developer-FAQ#i-have-a-new-function-or-feature-id-like-to-add-to-pytorch-should-it-be-in-pytorch-core-or-a-library-like-torchvision

In particular, this work might be a bit too new to be considered to be added into core just now. And a separate library sounds like the best place for it.
Don't hesitate to let us know if there is any low level feature that you're missing to implement this though!
